### DISCUSSION TITLE
Can I use the ethernet libraray with a different SPI instance?

### DISCUSSION THREAD

**Discussion by PerennialNovice** [stm32duino/Arduino_Core_STM32#1945] [Libraries]
Hi,

I fancy using both hardware SPI modules for different things and therefore would need to operate the ethernet library on a SPI instance that is not the standard Arduino SPI instance that gets created automatically by the SPI library.
Is this feasible without having to modify the library by myself (and thereby loosing all "automatic" updates)?

**Comment by fpistm** [comment]
If you talk about this one:
https://github.com/arduino-libraries/Ethernet

I guess no as it uses the default `SPI` instance, this is a recurrent issue when library does not allow to provide instance to use.
Anyway you should be able to use both SPI module on the same SPI instance while CS(SS) are different.

**Comment by PerennialNovice** [comment]
Hi fpistm,

thanks for the quick answer!

I know about the possibility to use different CS lines on the same instance, just thought it would be a nice idea to separate things a bit more in case "something" happens on the bus...

**Reply by fpistm** [reply]
Agreed. So you probably have to modify library.
